src/wire/libevent_callbacks.cpp

```diff
@@ -127,6 +128,29 @@ void StateMachine(LibeventSocket *conn) {
 
       case CONN_PROCESS: {
         bool status;
+
+        if(conn->pkt_manager.ssl_sent) {
+            // start SSL handshake
+            // TODO: consider free conn_SSL_context
```

Comment:
Thanks for your advice. We added code to destroy the object when the connection is closed.